frontend/src/metabase/api/utils/settings.ts

```diff
@@ -0,0 +1,41 @@
+import { useMemo } from "react";
+
+import type { Settings } from "metabase-types/api";
+
+import { useGetSettingsQuery } from "../session";
+import {
+  useGetAdminSettingsDetailsQuery,
+  useUpdateSettingMutation,
+} from "../settings";
+
+/**
+ * One hook to get setting values and mutators for a given setting
+ */
+export const useAdminSetting = <SettingName extends keyof Settings>(
+  settingName: SettingName,
+) => {
+  const {
+    data: settings,
+    isLoading: settingsLoading,
+    ...apiProps
+  } = useGetSettingsQuery();
+  const { data: settingsDetails, isLoading: detailsLoading } =
+    useGetAdminSettingsDetailsQuery();
+  const [updateSetting, updateSettingResult] = useUpdateSettingMutation();
```

Comment:
We could wrap this and pre-populate the setting key

Reply:
I went back and forth on this, but it actually ends up being more useful in the calling code to expose the general mutator

Reply:
How so? Just so that we don't need to type `value`?

Reply:
in a lot of places we're manipulating multiple settings, and it's nice to just have one setter function rather than juggling a few